### PR TITLE
fix(precompile): Patch precompile architecture in EVM

### PR DESCRIPTION
**Pull request by calbera**
- This PR fixes the staking precompile 
- Future PRs will clean up other precompiles

### CODE REVIEW

**Comment by itsdevbear** [comment]
closes #447

**Comment by calbera** [comment]
patch related to deeper bug #491

File: cosmos/testing/integration/rpc/eth_test.go [hunk 1]
```diff
@@ -47,19 +46,9 @@ func TestRpc(t *testing.T) {
...
-		_, err := net.WaitForHeightWithTimeout(1, defaultTimeout)
-		Expect(err).ToNot(HaveOccurred())
-
-		// Dial an Ethereum RPC Endpoint
-		rpcClient, err = gethrpc.DialContext(ctx, net.Validators[0].APIAddress+"/eth/rpc")
-		Expect(err).ToNot(HaveOccurred())
-		client = ethclient.NewClient(rpcClient)
-		Expect(err).ToNot(HaveOccurred())
+		net, client = StartPolarisNetwork(GinkgoT())
+		_ = net
```
Comment (itsdevbear) [inline cosmos/testing/integration/rpc/eth_test.go:51]: Remove _ thing

File: cosmos/testing/integration/utils/utils.go [hunk 2]
```diff
@@ -36,9 +37,27 @@ import (
 )
 
 const (
-	defaultTimeout = 10 * time.Second
+	DefaultTimeout = 100 * time.Second
```
Comment (itsdevbear) [inline cosmos/testing/integration/utils/utils.go:40]: Loewe timeout

File: cosmos/testing/network/network.go [hunk 3]
```diff
@@ -120,8 +121,8 @@ func DefaultConfig() network.Config {
 		TimeoutCommit:   2 * time.Second, //nolint:gomnd // 2 seconds is the default.
 		ChainID:         "polaris-2061",
 		NumValidators:   1,
-		BondDenom:       sdk.DefaultBondDenom,
-		MinGasPrices:    fmt.Sprintf("0.00006%s", sdk.DefaultBondDenom),
+		BondDenom:       "abera",
```
Comment (itsdevbear) [inline cosmos/testing/network/network.go:124]: Remove Bera change, no real point

File: cosmos/testing/network/network.go [hunk 4]
```diff
@@ -146,13 +149,21 @@ func BuildGenesisState() map[string]json.RawMessage {
...
-		// TODO MAKE CONFIGURABLE EVM DENOM
-		Coins: sdk.NewCoins(sdk.NewCoin("abera", sdk.NewInt(megamoney))),
+		Coins:   sdk.NewCoins(sdk.NewCoin("abera", sdk.NewInt(megamoney))),
 	})
 	genState[banktypes.ModuleName] = encoding.Codec.MustMarshalJSON(&bankState)
+
+	// Staking module
+	var stakingState stakingtypes.GenesisState
+	encoding.Codec.MustUnmarshalJSON(genState[stakingtypes.ModuleName], &stakingState)
+	stakingState.Params.BondDenom = "abera"
```
Comment (itsdevbear) [inline cosmos/testing/network/network.go:165]: Remove this change to Bera no real point.

**Review by itsdevbear — CHANGES_REQUESTED** [review]

File: contracts/src/cosmos/precompile/examples/Deploy.s.sol [hunk 5]
```diff
@@ -25,19 +25,38 @@
...
+import "../../../../lib/forge-std/src/Script.sol";
+import "../../../../lib/forge-std/src/console2.sol";
+import "../staking.sol";
+import "./LiquidStaking.sol";
+
+contract Deploy is Script {
+    address precompile = address(0xd9A998CaC66092748FfEc7cFBD155Aae1737C2fF);
+    IStakingModule staking = IStakingModule(precompile);
+
+    // TODO: script is broken because it runs its own evm. Fix Foundry.
```
[the remark this answers is not in the document]
Comment (calbera) [reply contracts/src/cosmos/precompile/examples/Deploy.s.sol:37]: yeah still broken

**Review by itsdevbear — APPROVED** [review]

**Pull request merged**